jupyterhub/values.yaml

```diff
@@ -55,6 +55,10 @@ hub:
     requests:
       cpu: 200m
       memory: 512Mi
+  containerSecurityContext:
+    runAsUser: *hubUid
```

Comment:
Oh wait hmmm...

Is `.Values.hub.uid` only used to create this? Then I think we should try deprecate hub.uid in favor of this configuration.

Reply:
OK I agree on just hard-coding `runAsUser: 1000` in values.yaml since you're correct that is the only thing using that value in the upstream helm chart. I suppose we wouldn't know if anyone is using `hub.uid` elsewhere downstream, but that's what the deprecation note is for.

One question on the deprecation note, I'm doing something like this in NOTES.txt:

```
{{- if .Values.hub.uid }}

DEPRECATION: hub.uid is deprecated in jupyterhub chart 0.9. Set the hub.containerSecurityContext.runAsUser value
directly instead.

{{- end }}
```

If we remove `hub.uid` from values.yaml and say it's deprecated, isn't that kind of a lie? Meaning shouldn't we leave `hub.uid` in values.yaml albeit deprecated and don't use it in the upstream helm chart, but give that warning if it's used downstream? Then when the deprecation period is over we could remove `hub.uid` from values.yaml and that might cause downstream builds to fail, but that's the intent.

Reply:
> Meaning shouldn't we leave `hub.uid` in values.yaml albeit deprecated and don't use it in the upstream helm chart, but give that warning if it's used downstream? Then when the deprecation period is over we could remove `hub.uid` from values.yaml and that might cause downstream builds to fail, but that's the intent.

Now I see that in practice this would be difficult because if we leave the deprecated `hub.uid` in values.yaml you'll always get a warning:

```console
$ helm template jupyterhub --set proxy.secretToken=nobodycares --notes
<snip>
DEPRECATION: hub.uid is deprecated in jupyterhub chart 0.9. Set the hub.containerSecurityContext.runAsUser value
directly instead.
---
```